src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -411,9 +457,29 @@ void ImageReplayer<I>::handle_bootstrap(int r) {
     return;
   }
 
+  bool do_resync = false;
+  {
+    Mutex::Locker locker(m_lock);
+
+    r = m_local_image_ctx->journal->check_resync_requested(&do_resync);
+    if (r < 0) {
+      derr << "failed to check if a resync was requested" << dendl;
+    }
+  }
+
+  if (do_resync) {
+    dout(5) << "image resync was requested" << dendl;
+    resync_image();
+    return;
+  }
+
   {
     Mutex::Locker locker(m_lock);
 
+    m_local_image_ctx->journal->add_listener(
```

Comment:
Now that you mention that possible race, I cannot check for the resync here, because the journaler will only be initialized later, and even if I add the listener before checking for resync it will not make any difference. I should add the listener as soon as the journaler is initialized, and then check for a resync,